File: 121.best-time-to-buy-and-sell-stock.cpp
// @leet start
class Solution {
public:
  int maxProfit(vector<int> &prices) {
    int l = 0;
    int r = 0;
    int answer = 0;
    while (r < prices.size()) {
      if (< prices[r] ) {
        answer = max(answer, prices[r] - prices[l]);
      } else {
        l = r;
      }
      r++;
    }
    return answer;
  }
};
// @leet end

File: 121.best-time-to-buy-and-sell-stock.ts
// @leet start
function maxProfit(prices: number[]): number {
	let l: number = 0;
	let r: number = 0;
	let soln: number = 0;
	while (r < prices.length) {
		if (prices[l] > prices[r]) l = r;
		if (prices[r] - prices[l] > soln) soln = prices[r] - prices[l];
		r++;
	}
	return soln;
}
// @leet end
